va/validation-authority.go

```diff
@@ -281,7 +270,9 @@ func (va *ValidationAuthorityImpl) fetchHTTP(identifier core.AcmeIdentifier, pat
 				return fmt.Errorf("Invalid port number in redirect")
 			}
 		} else if strings.ToLower(req.URL.Scheme) == "https" {
-			reqPort = "443"
+			reqPort = strconv.Itoa(va.httpsPort)
+		} else {
+			reqPort = strconv.Itoa(va.httpPort)
```

Comment:
@jsha should these also be 443 and 80 regardless of `va.httpPort`? e.g., if our `va.httpPort` is 5555 and `http://foo:5555/` redirects to `http://bar/`, should we connect to `bar:80`, or `bar:5555`?